integration-tests/dataproc_test_case.py

```diff
@@ -0,0 +1,143 @@
+import json
+import logging
+import random
+import unittest
+import subprocess
+from threading import Timer
+
+import os
+
+DEFAULT_TIMEOUT = 10  # minutes
+
+
+class DataprocTestCase(unittest.TestCase):
+    DEFAULT_ARGS = {
+        "SINGLE": [
+            "--single-node",
+            "--worker-machine-type n1-standard-4",
+            "--master-machine-type n1-standard-4",
+        ],
+        "STANDARD": [
+            "--num-masters 1",
+            "--num-workers 2",
+            "--worker-machine-type n1-standard-4",
+            "--master-machine-type n1-standard-4",
+        ],
+        "HA": [
+            "--num-masters 3",
+            "--num-workers 2",
+            "--worker-machine-type n1-standard-4",
+            "--master-machine-type n1-standard-4",
+        ]
+    }
+
+    COMPONENT = None
+    INIT_ACTION = None
+    TEST_SCRIPT_FILE_NAME = None
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        assert cls.COMPONENT
+        assert cls.INIT_ACTION
+
+    def createCluster(self, configuration, init_action, dataproc_version, metadata=None, scopes=None, properties=None,
+                      timeout_in_minutes=None):
+        self.name = "test-{}-{}-{}-{}".format(
+            self.COMPONENT,
+            configuration.lower(),
+            dataproc_version.replace(".", "-"),
+            random.randint(1, 10000),
```

Comment:
Use date instead of random here for better tracking.

Reply:
@aniket486 good point - by now I used YYYYMMDDHHmmSS - it has advantage over YYYYMMDD that you can run several clusters on the same day. WDYT?